plugins/woocommerce-blocks/assets/js/blocks/checkout/edit.tsx

```diff
@@ -69,6 +68,38 @@ export const Edit = ( {
 		showFormStepNumbers = false,
 	} = attributes;
 
+	const { requirePhoneField, showPhoneField } = useSelect( ( select ) => {
+		const phoneField = select( coreStore ).getEditedEntityRecord(
+			'root',
+			'site',
+			undefined
+		)?.woocommerce_checkout_phone_field;
+		return {
+			requirePhoneField: phoneField === 'required',
+			showPhoneField: phoneField !== 'hidden',
+		};
+	}, [] );
+
+	const setShowPhoneFieldEntity = ( showField: boolean ) => {
+		dispatch( coreStore ).editEntityRecord( 'root', 'site', undefined, {
+			woocommerce_checkout_phone_field: showField ? '' : 'hidden',
+		} );
+		// Hack to get it update in editor
+		wcSettings.defaultFields.phone.hidden = ! showField;
+		if ( showField ) {
+			wcSettings.defaultFields.phone.required = true;
+		}
+	};
```

Comment:
This logic was working fine before a quick clean up for this PR 🤔 needs a way to force the address component to remount.

Reply:
I think it was working fine in React 18, after pushing this PR, I changed back to React 17